operator-common/src/main/java/io/strimzi/operator/common/operator/resource/ReconcileResult.java

```diff
@@ -61,8 +60,8 @@ public String toString() {
     }
 
     /** No action was performed. */
-    public static final <P> ReconcileResult<P> noop() {
-        return NOOP;
+    public static final <P> ReconcileResult<P> noop(P resource) {
+        return new Noop<>(resource);
```

Comment:
why `noop` is now bringing the resource where no changes happen? out of curiosity ...

Reply:
We use the results on reconcile() in compose chains. Noop having a null result means we have to check for noop and use the desired resource. Noop having a proper result therefore simplifies those call sites.

Reply:
yeah, much better.